middle_end/flambda2/simplify/simplify_apply_expr.ml

```diff
@@ -1170,14 +1162,58 @@ let simplify_c_call ~simplify_expr dacc apply ~callee_ty ~arg_types ~down_to_up
       record_free_names_of_apply_as_used dacc ~use_id ~exn_cont_use_id apply
     in
     down_to_up dacc
-      ~rebuild:(rebuild_c_call apply ~use_id ~exn_cont_use_id ~return_arity)
+      ~rebuild:(rebuild_non_ocaml_function_call apply ~use_id ~exn_cont_use_id)
   | Invalid ->
     let rebuild uacc ~after_rebuild =
       let uacc = UA.notify_removed ~operation:Removed_operations.call uacc in
       EB.rebuild_invalid uacc (Closure_type_was_invalid apply) ~after_rebuild
     in
     down_to_up dacc ~rebuild
 
+let simplify_effect_op dacc apply (op : Call_kind.Effect.t) ~down_to_up =
+  fail_if_probe apply;
+  let denv = DA.denv dacc in
+  (match op with
+  | Perform { eff } -> DE.check_simple_is_bound denv eff
```

Comment:
Why don't we simplify thoses simples? For instance, if they could be an alias, or a constant.

Reply:
Ah that's just an oversight

Reply:
Fixed